### PR TITLE
Allow \r to precede \n for newlines.

### PR DESCRIPTION
**Pull request by hfern**
This allows ivy to be used on windows.

### CODE REVIEW

File: scan/scan.go [hunk 1]
```diff
@@ -279,7 +279,14 @@ func lexAny(l *Scanner) stateFn {
 	switch r := l.next(); {
 	case r == eof:
 		return nil
-	case r == '\n': // TODO: \r
+	case r == '\r':
+		// will also consume a following \n rune
+		if l.peek() == '\n' {
+			l.next()
+		}
+		l.emit(Newline)
```
Comment (robpike) [inline scan/scan.go]: You can just 'fallthrough'.

File: scan/scan.go [hunk 2]
```diff
@@ -279,7 +279,14 @@ func lexAny(l *Scanner) stateFn {
 	switch r := l.next(); {
 	case r == eof:
 		return nil
-	case r == '\n': // TODO: \r
+	case r == '\r':
+		// will also consume a following \n rune
```
Comment (robpike) [inline scan/scan.go]: change to

// Handle \r\n.

**Comment by hfern** [comment]
Both issues fixed

**Comment by robpike** [comment]
I fixed this another more general way, I believe.

**Pull request closed without merging**